test/Microsoft.AspNetCore.HostFiltering.Tests/HostFilteringMiddlewareTests.cs

```diff
@@ -179,5 +183,58 @@ public class HostFilteringMiddlewareTests
             var response = await server.CreateRequest("/").GetAsync();
             Assert.Equal(400, (int)response.StatusCode);
         }
+
+        [Fact]
+        public async Task SupportsDynamicOptionsReload()
+        {
+            var config = new ConfigurationBuilder().Add(new ReloadableMemorySource()).Build();
+            config["AllowedHosts"] = "localhost";
+            var currentHost = "otherHost";
+
+            var builder = new WebHostBuilder()
+                .ConfigureServices(services =>
+                {
+                    services.AddHostFiltering(options =>
+                    {
+                        options.AllowedHosts = new[] { config["AllowedHosts"] };
+                    });
+                    services.AddSingleton<IOptionsChangeTokenSource<HostFilteringOptions>>(new ConfigurationChangeTokenSource<HostFilteringOptions>(config));
```

Comment:
This won't be needed once the other PR is in right?

Reply:
Or is this testing a different API path?

Reply:
This mimics what happens in CreateDefaultBuilder, but we can't call that from this repo.